### PR TITLE
[Fuzzer] Generated columns now work properly with query-level aliases

### PR DESCRIPTION
**Pull request by Tishj**
This PR fixes (28) of #4978

The problem is that generated columns serialize expressions, containing ColumnRefExpressions, which contain a `column_names` vector, referencing columns of the table.

But in queries we can provide aliases for these columns, if those aliases are not forwarded to the expression, the expression will not resolve properly or worse, can become self-referencing (and cause a stack overflow attempting to resolve) as is the case illustrated in the mentioned issue.

We now look up the alias of every column referenced in the expression and replace it in the `column_names` of the ColumnRefExpression

### CODE REVIEW

File: test/fuzzer/pedro/gcol_stack_overflow.test [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+# group: [pedro]
+
+statement ok
+pragma enable_verification
+
+statement ok
+CREATE TABLE t0 (c2 AS (c0) VIRTUAL, c0 INT);
+
+query I
+SELECT 1 FROM t0 t2(c0) WHERE c0 = 0;
```
Comment (Mytherin) [inline test/fuzzer/pedro/gcol_stack_overflow.test:11]: Does this also work if we alias the table to its own name (e.g. t0 AS t0(c0))

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Looks good - one comment.

File: test/fuzzer/pedro/gcol_stack_overflow.test (hunk 1 above)
Comment (Tishj) [reply test/fuzzer/pedro/gcol_stack_overflow.test:11]: Will add more tests for that - but it should, as I already had a step to bake the table name into the expression during binding

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**